system/lib/wasmfs/wasmfs.cpp

```diff
@@ -101,6 +102,46 @@ __wasi_errno_t __wasi_fd_close(__wasi_fd_t fd) {
   return __WASI_ERRNO_SUCCESS;
 }
 
+long __syscall_fstat64(long fd, long buf) {
+  // Release FileTable lock after accessing desired open file.
+  std::shared_ptr<File> file = [&] {
+    auto fileTable = FileTable::get();
+    return fileTable[fd]->get().getFile();
+  }();
+
```

Comment:
The `->get().getFile()` stuff seems redundant to me, as a reader of the code (both the `->get()` and the `getFile()`). Can the FileTable simply return an `std::shared_ptr<File>`?